src/gt4py/backend/gtc_backend/dace/backend.py

```diff
@@ -348,6 +350,36 @@ def apply(cls, gtir: gtir.Stencil, sdfg: dace.SDFG, module_name: str, *, backend
         return formatted_code
 
 
+class DaCePyExtModuleGenerator(PyExtModuleGenerator):
+    def generate_imports(self):
+        res = super().generate_imports()
+        return res + "\nimport dace\nimport copy"
+
+    def generate_class_members(self):
+        res = super().generate_class_members()
+        filepath = self.builder.module_path.joinpath(
+            os.path.dirname(self.builder.module_path),
+            self.builder.module_name + "_pyext_BUILD",
+            self.builder.module_name + ".sdfg",
+        )
+        res += """
+_sdfg = None
+
+def __new__(cls, *args, **kwargs):
+    res = super().__new__(cls, *args, **kwargs)
+    cls._sdfg = dace.SDFG.from_file('{filepath}')
+    return res
+
+@property
+def sdfg(self) -> dace.SDFG:
+    return copy.deepcopy(self._sdfg)
```

Comment:
You're a better judge of this than I am not having written the code, but from the outside it seems non-performant to `deepcopy` a potentially huge nested structure every time a property is accessed.

Reply:
Yes, they are a pipeline performance issue. However, it's currently unsafe to use the same graph instance in wrapping. Tal is looking into improving this.